app/main/views/users.py

```diff
@@ -87,42 +93,74 @@ def download_users(framework_slug):
-    buyer_headings = [
+    users = {user["id"]: dict(user, briefs=[]) for user in data_api_client.find_users_iter(role="buyer")}
+
+    # join users with briefs (a "hash join")
+    for brief in data_api_client.find_briefs_iter(with_users=True):
+        for user in brief["users"]:
+            users[user["id"]]["briefs"].append(brief)
+
+    timestamp = datetime.utcnow().strftime('%Y%m%dT%H%M%S')
+    user_verbatim_fields = (
         "name",
         "emailAddress",
         "phoneNumber",
-        "createdAt",
-        "briefs"
-    ]
-
-    buyers_dict = {}
-    for buyer in buyers:
-        buyer['briefs'] = []
-        buyers_dict[buyer['id']] = buyer
-
-    for brief in briefs:
-        for user in brief['users']:
-            brief_string = '{} - {} - {}'.format(
-                brief['title'],
-                brief.get('location'),
-                'open' if brief['status'] == 'live' else brief['status'],
+    )
+    user_generated_fields = OrderedDict((
+        (
+            "createdAtDate",
+            lambda brief: brief.get("createdAt", "").partition("T")[0],  # cheap truncation of iso timestamp
+        ),
+    ))
+    brief_verbatim_fields = (
+        "id",
+        "title",
+        "location",
+        "lotSlug",
+    )
+    brief_generated_fields = OrderedDict((
+        (
+            "status",
+            lambda brief: "open" if brief.get("status") == "live" else brief.get("status", ""),
+        ),
+        (
+            "applicationsClosedAtDateIfClosed",
+            lambda brief:
+                (brief.get("applicationsClosedAt", "") if brief.get("status") == "closed" else "").partition("T")[0],
+        ),
+    ))
+
+    rows_iter = chain(
+        (
+            # header row
+            tuple(chain(
+                ("user.{}".format(field_name) for field_name in user_verbatim_fields),
+                ("user.{}".format(field_name) for field_name in iterkeys(user_generated_fields)),
+                ("brief.{}".format(field_name) for field_name in brief_verbatim_fields),
+                ("brief.{}".format(field_name) for field_name in iterkeys(brief_generated_fields)),
+            )),
+        ),
+        (
+            # data rows
+            tuple(chain(
+                (user.get(field_name, "") for field_name in user_verbatim_fields),
+                (func(user) for func in itervalues(user_generated_fields)),
+                (brief.get(field_name, "") for field_name in brief_verbatim_fields),
+                (func(brief) for func in itervalues(brief_generated_fields)),
+            ))
+            for user, brief in chain.from_iterable(
+                (
+                    (user, brief) for brief in
+                    (user["briefs"] or ({},))   # if user has an empty seq of briefs add a single fake blank
+                                                # one ("outer join" behaviour)
+                )
+                for user in sorted(itervalues(users), key=lambda user: user["name"])
```

Comment:
One thing it *does* make harder is commenting - not everywhere you want to leave a comment is a convenient/syntactically valid place to leave a comment, because it's all technically inside one expression.